tests/test_project/tests/test_password_history.py

```diff
@@ -100,3 +100,179 @@ def test_change_number_hasher_iterations(self):
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': 0
+        }
+    }])
+    def test_lookup_range_zero(self):
+        # If it's a 0, we're considering no passwords
+        self.create_user(1)
+        self.user_change_password(user_number=1, password_number=1)
+        self.user_change_password(user_number=1, password_number=2)
+        self.user_change_password(user_number=1, password_number=3)
+        self.assert_password_validation_True(user_number=1, password_number=1)
+        self.assert_password_validation_True(user_number=1, password_number=2)
+        self.assert_password_validation_True(user_number=1, password_number=3)
+        self.assert_password_validation_True(user_number=1, password_number=4)
+        self.user_change_password(user_number=1, password_number=4)
+        self.assert_password_validation_True(user_number=1, password_number=4)
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': -3
+        }
+    }])
+    def test_lookup_range_invalid(self):
+        # If it's negative, defaults to all the passwords
+        self.create_user(1)
+        self.user_change_password(user_number=1, password_number=1)
+        self.user_change_password(user_number=1, password_number=2)
+        self.user_change_password(user_number=1, password_number=3)
+        self.assert_password_validation_False(user_number=1, password_number=1)
+        self.assert_password_validation_False(user_number=1, password_number=2)
+        self.assert_password_validation_False(user_number=1, password_number=3)
+        self.assert_password_validation_True(user_number=1, password_number=4)
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': 2
+        }
+    }])
+    def test_delete_history_in_lookup_range(self):
+        # Older passwords outside of lookup_range are deleted
+        self.create_user(1)
+        self.user_change_password(user_number=1, password_number=1)
+        self.assertEqual(PasswordHistory.objects.count(), 1)
+        self.user_change_password(user_number=1, password_number=2)
+        self.assertEqual(PasswordHistory.objects.count(), 2)
+        self.user_change_password(user_number=1, password_number=3)
+        self.assertEqual(PasswordHistory.objects.count(), 2)
+
+    @override_settings(AUTH_PASSWORD_VALIDATORS=[{
+        'NAME': 'django_password_validators.password_history.password_validation.UniquePasswordsValidator',
+        'OPTIONS': {
+            'lookup_range': 2
+        }
+    }])
+    def test_delete_history_in_lookup_range(self):
```

Comment:
@spokerman12, duplicate test case name.

Reply:
Thanks! Renamed to `test_delete_history_by_date`